File: common/src/main/java/generations/gg/generations/core/generationscore/api/player/PlayerDataExtension.java
package generations.gg.generations.core.generationscore.api.player;

import com.google.gson.JsonObject;
import org.jetbrains.annotations.NotNull;

public abstract class PlayerDataExtension implements com.cobblemon.mod.common.api.storage.player.PlayerDataExtension {
    @NotNull
    @Override
    public JsonObject serialize() {
        var json = new JsonObject();
        json.addProperty(PlayerDataExtension.Companion.getNAME_KEY(), name());
        return json;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/api/player/PlayerDataExtension.kt
package generations.gg.generations.core.generationscore.common.api.player

import com.cobblemon.mod.common.api.storage.player.PlayerDataExtension
import com.google.gson.JsonObject

abstract class PlayerDataExtension : PlayerDataExtension {
    override fun serialize(): JsonObject {
        val json = JsonObject()
        json.addProperty(PlayerDataExtension.NAME_KEY, name())
        return json
    }
}
